tests/test_numbers.py

```diff
@@ -124,7 +127,7 @@ def test_scientific_notation(self):
         self.assertEqual(fmt, '1.2E3')
         # Exponent grouping
         fmt = numbers.format_scientific(12345, '##0.####E0', locale='en_US')
-        self.assertEqual(fmt, '12.345E3')
+        self.assertEqual(fmt, '1.2345E4')
```

Comment:
That is a deliberate change in output to fix what I think is a bug. The `##0.####E0` pattern implies only the first digit before the decimal separator is required.